scripts/output_code_size.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+SCRIPT_DIRECTORY="$(dirname "$(realpath "$0")")"
+. ${SCRIPT_DIRECTORY}/helper_find_llvm_install.sh
+
+OBJDUMP=$(find_llvm_tool llvm-objdump)
+
+if [ ! -x ${OBJDUMP} ] ; then
+	echo Unable to locate llvm-objdump, please set TOOLS_PATH to the directory containing the LLVM toolchain.
+	exit 1
+fi
+
+print_compartment_size() {
+	# Print all sections that match passed compartment name (potentially
+	# multiple ones, as compartments covers code and various data
+	# sections), and sum their sizes.
+	#
+	# Note that this does not validate compartment names. An invalid
+	# compartment name may return either 0kB, or an arbitrary number if the
+	# provided name matches legitimate compartments. This would be nice to
+	# fix at some point.
+	SIZE=$(${OBJDUMP} --headers $1 | grep -i $2 |
+		awk '{ sum += "0x"$3 } END { print (sum / 1024) "kB"}')
+	echo "Size of compartment '$2': ${SIZE}"
+}
+
+print_full_code_size() {
+	# Print all sections between the start of the firmware (`loader_start`)
+	# and the end of `__cap_relocs`, and sum the sizes of sections
+	SIZE=$(${OBJDUMP} --headers $1 |
+		awk '/loader_start/{f=1} /__cap_relocs/{f=0;print} f' |
+		awk '{ sum += "0x"$3 } END { print (sum / 1024) "kB"}')
+	echo "Size of the full binary: ${SIZE}"
```

Comment:
This doesn't need to be separate, just put it in the `END` block of the awk program.